### DISCUSSION TITLE
Do you have a translation into Portuguese(BR)?

### DISCUSSION THREAD

**Discussion by mrpunkdasilva** [diasurgical/DevilutionX#3240] [Q&A] [answered]
I want to know if it has a translation into Portuguese (BR) or just Portuguese?

**Comment by licaon-kter** [comment, marked as answer]
maybe https://github.com/diasurgical/devilutionX/tree/master/Translations

**Reply by mrpunkdasilva** [reply]
OK, thanks

**Reply by AJenbo** [reply]
The translation is 12% done, much of what is missing already have proposals that just need some one who speak Brasilien Portuguese to go over and remove the "needs work" flag if the line is ok. It would be a big help if you can download poedit and do so, and then upload the updated translation here.

**Reply by mrpunkdasilva** [reply]
> The translation is 12% done, much of what is missing already have proposals that just need some one who speak Brasilien Portuguese to go over and remove the "needs work" flag if the line is ok. It would be a big help if you can download poedit and do so, and then upload the updated translation here. 

I understand, I speak BR Portuguese (I'm from Brazil) I can help with the translation and proofreading

**Reply by AJenbo** [reply]
Sorry, it 88% done, only 12% needs review 😅

**Reply by AJenbo** [reply]
We will be releasing in a couple of days so please submit updates daily up until then so we can include as much as possible

**Reply by mrpunkdasilva** [reply]
Okay, thanks, I'll do it then

**Reply by pratagamer** [reply]
The new version has been released. in the changelog it says that support was added to pt-br, but I don't know how to change it in the game, it only stays in English even changing it in the .ini.

Need to download more files? Where are they to be downloaded?

**Reply by pratagamer** [reply]
> maybe https://github.com/diasurgical/devilutionX/tree/master/Translations

How to use this files?

**Reply by AJenbo** [reply]
In `diablo.ini` change `[Language] Code=en` to `[Language] Code=pt_BR`.

See here for where to locate the file: https://github.com/diasurgical/devilutionX/wiki/DevilutionX-diablo.ini-configuration-guide

**Reply by pratagamer** [reply]
Work, thanks!

**Reply by julealgon** [reply]
@AJenbo does `pt-BR` also work?

**Reply by AJenbo** [reply]
Not unless you rename the file ...

**Reply by mrpunkdasilva** [reply]
@AJenbo I was having this difficulty too, now I decided, thanks!

**Reply by julealgon** [reply]
> Not unless you rename the file ...

@AJenbo  Ok, just keep in mind that the actual standard uses an hyphen in the locale identifier (https://en.wikipedia.org/wiki/IETF_language_tag). 

> Each language tag is composed of one or more "subtags" separated by hyphens (-). Each subtag is composed of basic Latin letters or digits only.

If we don't support it, it might end up confusing a few more people when they change it using hyphens and it doesn't work.

Just figured I'd share in case you want to make this a bit more robust.

**Reply by AJenbo** [reply]
> @AJenbo Ok, just keep in mind that the actual standard uses an hyphen in the locale identifier (https://en.wikipedia.org/wiki/IETF_language_tag).

We follow the ISO/IEC 15897 standard, which is part of POSIX and thus more common with operating systems.

**Reply by StarHunter58** [reply]
> In `diablo.ini` change `[Language] Code=en` to `[Language] Code=pt_BR`.
> 
> See here for where to locate the file: https://github.com/diasurgical/devilutionX/wiki/DevilutionX-diablo.ini-configuration-guide

I renamed it and didn't work, the game continues in English

**Reply by AJenbo** [reply]
where did you get your version from?

**Reply by StarHunter58** [reply]
> where did you get your version from?

Mine is the one compatible with armx64

**Reply by AJenbo** [reply]
But where did you get it from? Is it a test build, did you compile it yourself og something else?

**Reply by StarHunter58** [reply]
> But where did you get it from? Is it a test build, did you compile it yourself og something else?

no bro, i got it from here: https://github.com/diasurgical/devilutionX/discussions/1403#discussioncomment-602730

**Reply by qndel** [reply]
that's 1.2.1 ... use latest or don't bother us

**Reply by StarHunter58** [reply]
I'm not bothering you, you answered because you wanted to

**Reply by AJenbo** [reply]
You do keep pinging us weather we answer or not... 1.2.1 does not have translation support. You need to use 1.3.0 or newer. Mind you even though you have an M1 CPU there the x86_64 build will works just fine on your computer, so there really isn't a reason to use an unofficial build.

https://github.com/diasurgical/devilutionX/releases/download/1.3.0/devilutionx-mac-x86_64.dmg

**Reply by StarHunter58** [reply]
Yes, I do have an M1 Mac, and I don't know nothing about Github. 
Well, I downloaded the recent version (1.3.0) and it works just as well as the "optimized" one. The translation is fine, but if you need help with some things I noticed, I would love to help.

**Reply by AJenbo** [reply]
Cool, the translation is here: https://github.com/diasurgical/devilutionX/blob/master/Translations/pt_BR.po

You can read about the translation process here: https://github.com/diasurgical/devilutionX/blob/master/Translations/glossary.md

And here: https://github.com/diasurgical/devilutionX/wiki/Translation-Guide

Keep in mind that a lot of improvements have already been made since 1.3.0 (1.4.0 will be coming soon), so some of the things you have noticed might already have been addressed and there might also be some new ones. So it might be a good idea to download a testbuild and test it with the latest translation.

**Reply by StarHunter58** [reply]
Keep up the great work, the game is running very well on the M1 and it seems native, as for the translation everything is ok, I just think there is a bug where the enemy names are still in English, but otherwise it is excellent and without grammatical errors , thanks for the attention.

**Reply by AJenbo** [reply]
They where probably not translated at the time. I believe they are translated in what will become 1.4.0.
